pkg/flowaggregator/flowaggregator.go

```diff
@@ -410,6 +418,7 @@ func (fa *flowAggregator) sendFlowKeyRecord(key ipfixintermediate.FlowKey, recor
 		fa.fillK8sMetadata(key, record.Record)
 		fa.aggregationProcess.SetMetadataFilled(record)
 	}
+	fa.fillPodLabels(key, record.Record)
```

Comment:
We should not do this before sending every flow record. Please optimize this.

Reply:
There is if-conditions in `fillPodLabels(key, record.Record)` function: only if current record doesn't have `sourcePodLabels` and `destinationPodLabels` field will do this filling job.

Reply:
I got confused because of these fields being added to the correlated field list and we are not making use of  `IsMetaDataFilled`. As these fields are added after processing the records in the intermediate process, not sure if we could tie the addition of these fields with `IsMetaDataFilled` flag or not. 
 
 We can further optimize the if condition to only one if check instead of two. As both of them are added together, we just have to check one of the fields. 
`GetInfoElementWithValue` is a costly function with O(n) complexity. Please check the ToT code in the IPFIX code. Can you think of a way to tie these fields to `isMetaDataFilled` flag?

Reply:
Thanks Srikar, `IsMetaDataFilled` is used for inter-node flow records which missing Pod information, but we are going to fill Pod labels for all kinds of records on the Flow Aggregator side, so we think combining these two conditions into `IsMetaDataFilled` flag is not appropriate.
I agree with other ideas: removed the labels from correlating list and do only one if check for them before filling.

Reply:
Sure. `IsMetaDataFilled` field is for existing fields in the flow record. As we may add some new fields like pod labels later in Flow Aggregator, could we enhance the`aggregatedFlowRecord` metadata to optimize the addition of those fields?

Reply:
Sure, a new filed like `HasFilledOnFlowAggregator` could be added, I will make the change on go-ipfix side.